dbms/cmake/version.cmake

```diff
@@ -63,6 +26,22 @@ execute_process(
   COMMAND_ERROR_IS_FATAL ANY
   )
 
+# Extract the major version number
+string(REGEX REPLACE "^v([0-9]+).*" "\\1" TIFLASH_VERSION_MAJOR ${TIFLASH_RELEASE_VERSION})
+# Extract the minor version number
+string(REGEX REPLACE "^v[0-9]+\\.([0-9]+).*" "\\1" TIFLASH_VERSION_MINOR ${TIFLASH_RELEASE_VERSION})
+# Extract the patch version number
+string(REGEX REPLACE "^v[0-9]+\\.[0-9]+\\.([0-9]+).*" "\\1" TIFLASH_VERSION_PATCH ${TIFLASH_RELEASE_VERSION})
+# Extract the extra information if it exists
+if (${TIFLASH_RELEASE_VERSION} MATCHES "-.*")
+  string(REGEX REPLACE "^v[0-9]+\\.[0-9]+\\.[0-9]+-(.*)$" "\\1" TIFLASH_VERSION_EXTRA ${TIFLASH_RELEASE_VERSION})
+  set(TIFLASH_VERSION "${TIFLASH_VERSION_MAJOR}.${TIFLASH_VERSION_MINOR}.${TIFLASH_VERSION_PATCH}-${TIFLASH_VERSION_EXTRA}")
+else ()
+  set(TIFLASH_VERSION "${TIFLASH_VERSION_MAJOR}.${TIFLASH_VERSION_MINOR}.${TIFLASH_VERSION_PATCH}")
+endif ()
+
+math (EXPR VERSION_REVISION "${TIFLASH_VERSION_PATCH} + ${TIFLASH_VERSION_MINOR}*100 + ${TIFLASH_VERSION_MAJOR}*10000")
```

Comment:
I don't think we should be using "revision" anymore. In old ClickHouse it indicates their build number, which is trivial in our versioning. So removing the usage of `VERSION_REVISION` would be ideal.

Reply:
Some more context: When we initialing formalize our versioning on top the old ClickHouse versioning since `4.1.0`, we treat their `revision` as our `patch`.

So there is actually another option: we keep treating `revision` as our `patch`, rather than introducing a whole new `patch` terminology.

Reply:
ok, I will remove `revision`

Reply:
@zanmato1984 maybe keep it to as a unique flag of a version, and then we can use it to check whether the version of client and server are matched.

Reply:
* In general, new version clinet should be able to connect to old version server. Unless there are incompatible protocol changes. Checking whether the version of client and server are strictly matched is meaningless.
 Instead of `revision`, the client can send its `TIFLASH_VERSION`,`GIT_HASH` to the server, and the server can log it down for some troubleshooting. That's all.
* The TCP client is a debug-only tool for TiFlash on some test cases and is disabled on prod env now. There should be no further changes to the TCP protocol.

IMO, if removing `revision` does not involve big code changes, then it is ideal to do so.

Reply:
removed